pandas/core/generic.py

```diff
@@ -5522,6 +5522,14 @@ def __finalize__(
                 object.__setattr__(self, name, getattr(other, name, None))
 
         if method == "concat":
+            # Issue #41828, retain the attrs only if all NDFrame have the same
+            # attrs.
+            attrs = other.objs[0].attrs
```

Comment:
is this hit if we have an empty input?

Reply:
@jreback I wonder what do you mean by empty input?

If is it empty dataframe or series like Dataframe() or Series(), the attr will be conserved when it should be conserved and lost when it should get lost.